### PR TITLE
feat: add endpoint destroy users

### PR DESCRIPTION
**Pull request by ayocodingit**
# Overview

- feat: add endpoint destroy users

cc: @jabardigitalservice/jds-backend 

## Evidence
- title: feat: add endpoint destroy users
- project: Desa Digital
- participants: @ayocodingit @rachadiannovansyah @khihadysucahyo @bangunbagustapa

### CODE REVIEW

File: src/modules/users/user_repository.ts [hunk 1]
```diff
@@ -39,4 +39,6 @@ export namespace User {
   }
 
   export const findById = async (id: string) => Query().where('users.id', id).first()
+
+  export const destroy = async (id: number) => Users().where('id', id).delete()
```
Comment (khihadysucahyo) [inline src/modules/users/user_repository.ts:43]: it seems that this function doesn't need to be async?
`destroy` already and only returns promise, u just need to call the function from the service with `await`

**Review by rachadiannovansyah — APPROVED** [review]

File: src/modules/users/user_repository.ts [hunk 2]
```diff
@@ -39,4 +39,6 @@ export namespace User {
   }
 
   export const findById = async (id: string) => Query().where('users.id', id).first()
```
Comment (khihadysucahyo) [inline src/modules/users/user_repository.ts:41]: this is also not required to be defined as async function

File: src/modules/users/user_repository.ts (hunk 1 above)
Comment (ayocodingit) [reply src/modules/users/user_repository.ts]: yes thank for your information, next we will remove async when return promise

**Review by khihadysucahyo — APPROVED** [review]

File: src/modules/users/user_repository.ts (hunk 1 above)
Comment (khihadysucahyo) [reply src/modules/users/user_repository.ts]: ngokhey, but it's worth pointing out, this only applies if in a function block there is only one promise that doesn't depend on each other

**Pull request merged**